### DISCUSSION TITLE
Rspam tries to load dkim keys for unconfigured domains

### DISCUSSION THREAD

**Discussion by TheQue42** [rspamd/rspamd#3869] [Q&A]
Hi, 

I just started using rspamd, since my opendkim seems to flag valid messages as bad signatures. After running for a few weeks with debugging for dkim on, I've found these logs:

```
Aug 25 20:03:42 balrog rspamd[19599]: <5800e5>; task; dkim_module_load_key_format: cannot load dkim key /var/rspamd/dkim/<work-domain>.se.dkim.key: cannot stat key file: '/var/rspamd/dkim/<domain>.se.dkim.key' No such file or directory
```
But my dkim_signing.conf only consists of:
```
#> cat dkim_signing.conf
domain {
    <private-domain>.se {
        path = "/var/db/dkim/<private-domain>.private";
        selector = "default";
    }
}
enabled = true;
```
Now, postfix was configured with relaying for the <work-domain> above, but there is not a single reference to <work-domain> anywhere in the rspamd config.

What have I misunderstood?

Thanks, 
TQ

**Comment by andryyy** [comment]
Try to add “ try_fallback = false;” to the conf file. 

André

**Reply by TheQue42** [reply]
Done, although I cant say I understand why :-) What does it cause?  Looking into "modules.d/dkim_signing.conf" it seems already set, and I haven't overridden it in my local.d-copy.

**Reply by fatalbanana** [reply]
@andryyy meant `try_fallback = false` I suppose.

Signing domains needn't be configured explicitly and keys will be used opportunistically if available on disk; it's described [here](https://rspamd.com/doc/modules/dkim_signing.html#principles-of-operation). Setting `try_fallback = false` disables that behaviour and calls for domains to be explicitly configured.

**Reply by andryyy** [reply]
Oh, absolutely. Thanks for correcting me!

**Reply by TheQue42** [reply]
Ah, it took me a while to realize that you've edited your response :-) Thanks for clarifying!
